ios/wrappers/utils/CioConfigUtils.swift

```diff
@@ -56,3 +56,62 @@ func cioInitializeConfig(from config: RCTCioConfig, logLevel: String?) -> CioCon
     
     return CioConfig(cdp: cdpConfigBuilder.build(), inApp: inAppConfig)
 }
+
+// To convert dictionary to Codable
+struct AnyCodable: Codable {
+    let value: Any
+    
+    init(_ value: Any) {
+        self.value = value
+    }
+    
+    init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        
+        if let value = try? container.decode(String.self) {
+            self.value = value
+        } else if let value = try? container.decode(Int.self) {
+            self.value = value
+        } else if let value = try? container.decode(Double.self) {
```

Comment:
When decoding, when you try to decode Int before Double, this could lead to precision loss for large numbers that are Doubles but can fit into Int.